torch/_inductor/pattern_matcher.py

```diff
@@ -1239,38 +1243,37 @@ def apply(self, graph: torch.fx.GraphModule) -> int:
                 get_mutation_region_id, graph
             )
         count = 0
-        # Copy the graph nodes so that modifications to the graph are not observed
-        # during the pass
-        for node in reversed(list(graph.nodes)):
+        for node in sorted(
```

Comment:
Should we refactor things to avoid calling `sorted()` twice in use cases like this?  Maybe we need another help that allows you combine multiple queries.

Reply:
Designing a clean API for combining multiple queries sounds hard.
```
find_nodes(ops=List[str], targets=List[Optional[Target]])
```
seems wrong since it forces passing None in targets. Perhaps I should add flag to disable sorting and also allow reserve sorting?

Reply:
Added a flag for sorting